ncm-gip2 14.10.0-rc1 improperly calls FileWriter

Blocking issue for 14.10.0 release (https://github.com/quattor/release/issues/58).

Running `ncm-gip2` produces the following error:

```
Use of uninitialized value $path in concatenation (.) or string at /usr/lib/perl/CAF/FileWriter.pm line 139.
[ERROR] component gip2 executing method Configure fails: Can't call method "verbose" on an undefined value at /usr/lib/perl/CAF/FileWriter.pm line 139.
```

It looks like something weird happens when calling FileWriter...

--- a/ncm-gip2/src/main/perl/gip2.pm
+++ b/ncm-gip2/src/main/perl/gip2.pm
@@ -23,7 +23,7 @@ use File::Basename;
 local(*DTA);
 
 sub write_encoded {
-    my ($self, $file, $perm, $contents);
+    my ($self, $file, $perm, $contents) = @_;
     my $fh = CAF::FileWriter->new($file, mode => $perm, log => $self);
     print $fh encode_utf8($contents);
     return $fh->close();

